### DISCUSSION TITLE
Slow loading

### DISCUSSION THREAD

**Discussion by Annovdbos** [Budibase/budibase#1882] [Help] [answered]
Hi,

I'm currently working with budibase. Really love the simplicity of building applications fast. However, often I have trouble with loading time. It often keeps loading and it is stuck there. Is there something I have to change on my PC settings? Or is it a bug? Could somebody help me with this issue?

Thanks in advance! :)

**Comment by shogunpurple** [comment, marked as answer]
@Annovdbos when you say slow loading, what do you mean?

Slow to load your builder? Your apps? What hardware are you running budibase on? Do you have other stuff running on this machine?

**Reply by Annovdbos** [reply]
Hi @shogunpurple 

The builder is loading slowly or sometimes it's just stuck. It doesn't load at all. I'm not that knowledgeable in the backend world (yet). But I believe I'm running it on docker, is that possible? 😅 

So, I do know it's locally hosted. And I also have MAMP running on this machine. Could be that those interfere with eachother perhaps? I believe I'm running them on different ports though. Budibase on 4001 and MAMP on 80. But not sure what that means exactly and if this could cause trouble in this department.

Thanks for the help!

**Reply by shogunpurple** [reply]
I don't think MAMP will interfere, as long as they are on separate ports, as you say.

What kind of hardware are you running budibase on? Budibase generally runs fairly comfortably on a digitalocean droplet with 2GB of memory. How much memory does your machine have?

**Reply by Annovdbos** [reply]
Okay, then I don't tink that will be the problem. 

I believe I'm running it on digital ocean indeed. I remember having downloaded that when I installed budibase. Is there a way to check? My machine (MacBook Pro) has 8 GB of memory.

**Reply by shogunpurple** [reply]
@Annovdbos You don't download digitalocean - it's a platform for hosting that you can use to host budibase. It seems like you are just running budibase on your local machine.

Docker on mac requires a VM to run. This uses a **ton** of memory, and the memory struggles of running docker on mac are well documented. Your machine is going to struggle to run the full budibase stack in docker, MAMP (which includes Apache, NGINX and MySQL). I'd imagine you are also using this machine for a bunch of other stuff like web browsing and other tasks - further stretching the memory.

For getting the best out of budibase, we recommend running budibase (or any other self hosted software) on its own server. The simplest way to do this right now is with our one click digitalocean droplet:

https://docs.budibase.com/self-hosting/hosting-methods/digitalocean

We are in the process of building out a cloud platform, which means you will not need to run/manage budibase infastructure yourself at all. You can just use the cloud platform like any other application.

**Reply by Annovdbos** [reply]
Ah that's the problem then. I indeed also use my mac for work and building websites.

I'll setup digitalocean to host budibase on. A cloud platform would absolutely be awesome!

Thanks for the quick help. Appreciate it! :)

**Comment by Annovdbos** [comment]
I have setup a digitalocean droplet now. But I can't find out how to run budibase on this server now. 

I can still access budibase via https://localhost:4001. But via this way it still shows the same problems as before. I think that's still hosted locally. However, how do I host the app on the digitalocean droplet? 

I already did ssh into the IP-address with the password. It ran succesfully, but I don't know how to access budibase?

**Reply by shogunpurple** [reply]
Hi @Annovdbos - go to the IP address of your digitalocean droplet in the browser:

`<droplet-ip>:10000`

**Reply by Annovdbos** [reply]
Ah great, that was the piece I was missing. Thanks! :)
